src/main/java/be/syntra/devshop/DevshopFront/configuration/WebConfig.java

```diff
@@ -55,16 +56,13 @@ public CartDto getCurrentCart() {
     }
 
     @Bean
+    @SessionScope
     public SearchModel getSearchModel() {
         return new SearchModel();
```

Comment:
Partly because all this happened in the frontEnd only (against the cache) and: each added filter or changed ordering takes into account the previous results, until the user returns to the startpage or resets the filters.  Therefore we need the previous state of the SearchModel, i can't think of a better way to solve this at the moment... will discusss with the team.

Reply:
Ok makes, sense. Not saying that you should change it, but an alternative could have been to use query parameters: `?query=blabla&minPrice=100&...` and capture them in the Controller to (re)create the SearchModel before sending it to the view. This solution works fine as well, but you just won't be able to bookmark or share a link with particular search filters this way.

Reply:
We'd be looking to change that as part of the next ticket. The refactoring was divided among several tickets to lighten the workload and gradually make the desired changes.